dev-packages/node-integration-tests/suites/tracing/socket.io/test.ts

```diff
@@ -0,0 +1,59 @@
+import { cleanupChildProcesses, createRunner } from '../../../utils/runner';
+
+describe('socket.io auto instrumentation', () => {
+  afterAll(() => {
+    cleanupChildProcesses();
+  });
+
+  test('should auto-instrument `socket.io` package', done => {
+    const SERVER_TRANSACTION = {
+      transaction: 'GET /',
+      spans: expect.arrayContaining([
+        expect.objectContaining({
+          origin: 'auto.socket.otel.socket_io',
+          data: {
+            'sentry.origin': 'auto.socket.otel.socket_io',
+            'messaging.destination': '/',
+            'messaging.destination_kind': 'topic',
+            'messaging.socket.io.event_name': 'test',
+            'messaging.socket.io.namespace': '/',
+            'messaging.system': 'socket.io',
+            'otel.kind': 'PRODUCER',
+            'sentry.op': 'message',
+          },
+          description: '/ send',
+          op: 'message',
+          status: 'ok',
+        }),
+      ]),
+    };
+
+    const CLIENT_TRANSACTION = {
+      transaction: 'test_reply receive',
+      contexts: {
+        trace: expect.objectContaining({
+          span_id: expect.any(String),
+          trace_id: expect.any(String),
+          data: expect.objectContaining({
+            'sentry.op': 'message',
+            'sentry.origin': 'auto.socket.otel.socket_io',
+            'otel.kind': 'CONSUMER',
+            'messaging.system': 'socket.io',
+            'messaging.destination': '/',
+            'messaging.operation': 'receive',
+            'messaging.socket.io.event_name': 'test_reply',
+          }),
+          origin: 'auto.socket.otel.socket_io',
+          op: 'message',
+          status: 'ok',
+        }),
+      },
+    };
+
+    createRunner(__dirname, 'scenario.js')
+      .expect({ transaction: SERVER_TRANSACTION })
+      .expect({ transaction: CLIENT_TRANSACTION })
```

Comment:
I'm not sure why the server socket and client socket are not part of the same transaction. In [scenario.js](https://github.com/getsentry/sentry-javascript/blob/93e0e53cb6911a7c3261cf74de215802ac3e1ad7/dev-packages/node-integration-tests/suites/tracing/socket.io/scenario.js#L30-L48), the server socket emits an event to the client, which triggers the client to send back to the server.

Reply:
I guess this is expected given the same behaviour in other [integrations](https://github.com/getsentry/sentry-javascript/pull/13714/files) where consumer and producer are separate.